plasmapy/physics/transport.py

```diff
@@ -107,98 +168,745 @@ def Coulomb_logarithm(T, n_e, particles, V=None):
+        roughly its reciprocal.
 
-    # The outer impact parameter is the Debye length.  At distances
-    # greater than the Debye length, the electrostatic potential of a
-    # single particle is screened out by the electrostatic potentials
-    # of other particles.  Past this distance, the electric fields of
-    # individual particles do not affect each other much.  This
-    # expression neglects screening by heavier ions.
+    Raises
+    ------
+    ValueError
+        If the mass or charge of either particle cannot be found, or
+        any of the inputs contain incorrect values.
 
-    T = T.to(units.K, equivalencies=units.temperature_energy())
+    UnitConversionError
+        If the units on any of the inputs are incorrect
+
+    UserWarning
+        If the inputted velocity is greater than 80% of the speed of
+        light.
 
-    b_max = Debye_length(T, n_e)
+    TypeError
+        If the n_e, T, or V are not Quantities.
 
-    # The choice of inner impact parameter is more controversial.
-    # There are two broad possibilities, and the expressions in the
-    # literature often differ by factors of order unity or by
-    # interchanging the reduced mass with the test particle mass.
+    Notes
+    -----
+    The Coulomb logarithm is given by
 
-    # The relative velocity is a source of uncertainty.  It is
-    # reasonable to make an assumption relating the thermal energy to
-    # the kinetic energy: reduced_mass*velocity**2 is approximately
-    # equal to 3*k_B*T.
+    .. math::
+        \ln{\Lambda} \equiv \ln\left( \frac{b_{max}}{b_{min}} \right)
 
-    # If no relative velocity is inputted, then we make an assumption
-    # that relates the thermal energy to the kinetic energy:
-    # reduced_mass*velocity**2 is approximately equal to 3*k_B*T.
+    where :math:`b_{min}` and :math:`b_{max}` are the inner and outer
+    impact parameters for Coulomb collisions _[1].
 
-    if V is None:
-        V = np.sqrt(3 * k_B * T / reduced_mass)
+    Examples
+    --------
+    >>> from astropy import units as u
+    >>> n = 1e19*u.m**-3
+    >>> T = 1e6*u.K
+    >>> particles = ('e', 'p')
+    >>> Coulomb_logarithm(T, n, particles)
+    14.748259780491056
+    >>> Coulomb_logarithm(T, n, particles, 1e6*u.m/u.s)
+    11.363478214139432
+
+    References
+    ----------
+    [1] Dense plasma temperature equilibration in the binary collision 
+    approximation. D. O. Gericke et. al. PRE,  65, 036418 (2002).
+    DOI: 10.1103/PhysRevE.65.036418
+    """
+    # boiler plate checks
+    T, masses, charges, reduced_mass, V = boilerPlate(T=T,
+                                                      particles=particles,
+                                                      V=V)
+    # Debye length
+    lambdaDe = Debye_length(T, n_e)
+    # ion sphere radius
+    # !!! this should probably use ion density instead of electron
+    # density. Another reason why we need average ionization!!!
```

Comment:
Agreed.

Reply:
Thoughts on how to handle this in the mean time?

Reply:
Probably the most optimal solution is to stave off work on this PR and get average ionization done, so we don't have to rewrite half of this one - and even rewriting half of this PR would be a huge undertaking.

Reply:
I added `z_mean` for recovering the ion density and calculating the ion sphere radius.